save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt

```diff
@@ -126,16 +280,159 @@ class ProjectView : RComponent<ProjectProps, ProjectViewState>() {
         div("row") {
             child(cardComponent {
                 div("text-center") {
-                    div("pb-4") {
-                        h6(classes = "d-inline") {
-                            +"Path to property file: "
+                    div("card shadow mb-4") {
+                        a(classes = "d-block card-header py-3") {
+                            attrs {
+                                onClickFunction = { setState { isProjectAsGitUrl = !isProjectAsGitUrl } }
+                            }
+                            h6("m-0 font-weight-bold text-primary") {
+                                +"Upload project as git url"
+                            }
+                        }
+                        // Collapse card to load git url
+                        div {
+                            attrs.classes = if (state.isProjectAsGitUrl) setOf("collapse", "show") else setOf("collapse")
+                            div("card-body") {
+                                div("pb-3") {
+                                    div("d-inline-block") {
+                                        h6(classes = "d-inline") {
+                                            +"Git url: "
+                                        }
+                                    }
+                                    div("d-inline-block ml-2") {
+                                        input(type = InputType.text, name = "itemText2") {
+                                            this.key = "itemText2"
+                                            attrs {
+                                                attrs.autoFocus = state.editableUrlInput
+                                                state.gitUrlProject?.let {
+                                                    value = it
+                                                } ?: run {
+                                                    props.executionRequest.gitRepository.url?.let {
+                                                        value = it
+                                                    }
+                                                }
+                                                placeholder = "https://github.com/"
+                                                onChangeFunction = {
+                                                    val target = it.target as HTMLInputElement
+                                                    setState {
```

Comment:
Form's content can be snapshotted only right before modal opening; this way we can restore it on modal close, but don't need to re-render on every inputted letter